modules/gke-hub/main.tf

```
@@ -38,6 +38,7 @@ resource "google_gke_hub_membership" "default" {
   provider      = google-beta
   for_each      = var.clusters
   project       = var.project_id
+  location      = var.location != null ? var.location : "global"
```

Don't try to guess. If the user wants global, let them pass it it.
```suggestion
  location      = var.location
```